src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -191,61 +201,133 @@ def find_subscriptions_on_login(self,
                 use_device_code = True
 
             if not use_device_code:
+                from azure.identity import CredentialUnavailableError
                 try:
-                    authority_url, _ = _get_authority_url(self.cli_ctx, tenant)
-                    subscriptions = subscription_finder.find_through_authorization_code_flow(
-                        tenant, self._ad_resource_uri, authority_url)
-                except RuntimeError:
+                    credential, auth_profile = self.login_with_interactive_browser(tenant)
+                except CredentialUnavailableError:
```

Comment:
I am not sure whether we should discard `in_cloud_console()` and `can_launch_browser()`, and rely on `CredentialUnavailableError` raised by `InteractiveBrowserCredential` totally.  @qianwens any thoughts?

Reply:
need test to check if you can handle the error properly to give good error message

---
In reply to: [409972764](https://github.com/Azure/azure-cli/pull/13053#discussion_r409972764) [](ancestors = 409972764)